icicle/include/fields/params_gen.cuh

```diff
@@ -0,0 +1,146 @@
+#pragma once
+#ifndef PARAMS_GEN_H
+#define PARAMS_GEN_H
+
+#include "storage.cuh"
+#include "host_math.cuh"
+
+using namespace host_math;
```

Comment:
I think its better to not use this so the functions need the namespace specifier and its easier to follow where the functions are coming from.

It would also allow us to remove the `_host` suffix on the function names